editoast/src/views/timetable/stdcm.rs

```diff
@@ -251,7 +255,38 @@ async fn stdcm(
 
     let stdcm_response = stdcm_request.fetch(core_client.as_ref()).await?;
 
-    // 6. Handle STDCM Core Response
+    // 6. Check if the current tracing level is debug or lower, and if so, log STDCM request and response
+    if tracing::level_filters::LevelFilter::current() >= tracing::Level::DEBUG {
+        let user_id = auth.authorizer()?.user_id();
```

Comment:
Do we want to bubble up the error if `authorizer()` fails? What about putting everything that is inside the `if` inside the closure. I would expect minimal impact on the code, something like this

```rust
if LevelFilter::current >= Level::DEBUG {
  tokio::spawn(async move { /* ... */ });
}
```

Reply:
`user_id` is not null in the DB, and we need to know which user made the request. So, I think it's important to throw an error if we can't retrieve `user_id`.

Reply:
Is it possible to short-circuit in production? @leovalais @woshilapin

Reply:
The gateway being the only exposed access point prevents that. But we may have unauthenticated, but legit, requests to editoast from behind it, on purpose.

In any case, I think the user_id should be nullable so that we can still log requests in development when authentication is disabled.